scripts/eurostat/regional_statistics_by_nuts/birth_death_migration/import_data.py

```diff
@@ -58,101 +80,141 @@ def preprocess_data(raw_df):
-        'Count_Death_AsAFractionOf_Count_Person', 'IncrementalCount_Person',
-        'GrowthRate_Count_Person', 'Count_Person', 'Count_BirthEvent'
-    ]
-    clean_df = clean_df[original_names]
-    clean_df[['GBIRTHRT', 'GDEATHRT',
-              'GROWRT']] = clean_df[['GBIRTHRT', 'GDEATHRT',
-                                     'GROWRT']].apply(pd.to_numeric)
-    clean_df[['GBIRTHRT', 'GDEATHRT',
-              'GROWRT']] /= 1000  # apply scaling factor of 1000
-    clean_df.columns = new_names
-    clean_df.to_csv(output_path, index=False)
-
-
-if __name__ == '__main__':
-    download_link = "https://ec.europa.eu/eurostat/estat-navtree-portlet-prod/BulkDownloadListing?file=data/demo_r_gind3.tsv.gz"
+    try:
+        logging.info('file cleaning started ')
+        num_clean_columns = len(preprocessed_df.columns) // 2 + 1
+        # drop unused columns
+        clean_df = preprocessed_df.iloc[:, :num_clean_columns]
+
+        # replace colon with NaN.
+        clean_df = clean_df.replace(':', '')
+        clean_df['geo'] = clean_df['geo'].apply(
+            lambda geo: f'nuts/{geo}' if any(geo.isdigit() for geo in geo) or
+            ('nuts/' + geo in NUTS1_CODES_NAMES) else COUNTRY_MAP.get(
+                geo, f'{geo}'))
+
+        # trim the space in the time column i.e. '2020 ' -> '2020'
+        clean_df['time'] = clean_df['time'].astype('int32')
+        original_names = [
+            'geo', 'time', 'DEATH', 'GBIRTHRT', 'GDEATHRT', 'GROW', 'GROWRT',
+            'JAN', 'LBIRTH'
+        ]
+        new_names = [
+            'geo', 'time', 'Count_Death',
+            'Count_BirthEvent_AsAFractionOfCount_Person',
+            'Count_Death_AsAFractionOfCount_Person', 'IncrementalCount_Person',
+            'GrowthRate_Count_Person', 'Count_Person', 'Count_BirthEvent'
+        ]
+        clean_df = clean_df[original_names]
+        clean_df[['GBIRTHRT', 'GDEATHRT',
+                  'GROWRT']] = clean_df[['GBIRTHRT', 'GDEATHRT',
+                                         'GROWRT']].apply(pd.to_numeric)
+        clean_df[['GBIRTHRT', 'GDEATHRT',
+                  'GROWRT']] /= 1000  # apply scaling factor of 1000
+        clean_df.columns = new_names
+        clean_df.to_csv(output_path, index=False)
+        logging.info('file cleaning completed ')
+    except Exception as e:
+        logging.error(f'cleaning error {e}')
+
+
+def main(_):
+    mode = _FLAGS.mode
+    download_link = "https://ec.europa.eu/eurostat/api/dissemination/sdmx/2.1/data/demo_r_gind3/?format=TSV&compressed=true"
     output_path = 'EurostatNUTS3_BirthDeathMigration.csv'
-    raw_df = download_data(download_link)
-    preprocessed_df = preprocess_data(raw_df)
-    clean_data(preprocessed_df, output_path)
+    input_path = os.path.join(_MODULE_DIR, 'input_files')
+    if not os.path.exists(input_path):
+        os.makedirs(input_path)
+    input_file = os.path.join(input_path, 'input_file.tsv')
+
+    if mode == "":
```

Comment:
Implement the main method as per Ajai's suggestion [here](https://github.com/datacommonsorg/data/pull/1105#discussion_r1832469518)